bazel/scripts/run_integ_tests.sh

```diff
@@ -474,17 +490,30 @@ then
     echo "#######################################"
     echo "EXTENDED TESTS"
     echo "#######################################"
+    EXTENDED_SETUP="success"
     if [[ "${SKIP_EXTENDED_SETUP_AND_TEARDOWN}" == "false" ]];
     then
-        setup_extended_tests
+        if setup_extended_tests;
+        then
+            EXTENDED_SETUP="success"
+        else
+            EXTENDED_SETUP="failed"
+            echo "The setup failed. Skipping extended tests."
+        fi
     fi
 
-    TEST_BATCH_TO_RUN=( "${EXTENDED_TEST_TARGETS[@]}" )
-    run_test_batch
+    if [[ "${EXTENDED_SETUP}" == "success" ]];
+    then
+        TEST_BATCH_TO_RUN=( "${EXTENDED_TEST_TARGETS[@]}" )
```

Comment:
Perhaps we can move this into line 498, then we do not need the `EXTENDED_SETUP` variable.

Reply:
If we move it there then the tests will not run if you add the `--skip-setup-teardown-extended` option.

Reply:
I missed that, got it